File: cpp/849_MaximizeDistanceToClosestPerson.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int left = 0, right = 0, highest = 0;
        
        while (right < seats.size() && seats[right] != 1) {
            right++;
            highest++;
        }
        for (int i = 0; i < seats.size(); ++i) {
            if (seats[i] == 1) {
                right = i + 1, left = i;
                while (right < seats.size() && seats[right] == 0) right++;
                if (right == seats.size()) return max(right - 1 - i, highest);
            } else {
                highest = max(min(right - i, i - left),  highest);
            }
        }
        return highest;
    }
};

File: python/849_MaximizeDistanceToClosestPerson.py
class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        left, right, highest, N = 0, 0, 0, len(seats)
        while right < N and seats[right] != 1:
            right += 1
            highest += 1
        
        for i, seat in enumerate(seats):
            if seat == 1:
                left, right = i, i + 1
                while right < N and seats[right] != 1:
                    right += 1
                if right == N: return max(highest, right - 1 - i)
            else:
                highest = max(highest, min(i - left, right - i))
        return highest
